server/main.cpp

```diff
@@ -127,6 +127,7 @@ int main()
             std::vector<int> playerPriorities(cse125constants::NUM_PLAYERS, 0); // index i stores player i's priority
             unsigned int priorityCtr = 1;
 
+            server.queueMtx.lock();
```

Comment:
Why are we locking here instead of line 147?

Reply:
I think just to make sure that there's no race condition when reading from the queue while clients are pushing to it. However, this might be less performant. Do you get any crashes if you put the lock on line 147?